getpaid/backends/pagseguro/__init__.py

```diff
@@ -0,0 +1,183 @@
+        dom = parseString(response)
+        
+        checkout = dom.getElementsByTagName("checkout")
+        if checkout:
+            childNodes = checkout[0].childNodes
+            code = childNodes[0].firstChild.nodeValue
+            date = childNodes[1].firstChild.nodeValue
+
+        self.payment.external_id = reference
+        self.payment.description = code
+        self.payment.save();
+        
+        return "%s%s" % (gateway_url2, self._TRANSACTION_INSTRUCTION_PAGE+code), 'GET', {}
+
+    @staticmethod
+    def process_notification(params):
+        _TRANSACTION_CONSULT_URL = "https://ws.pagseguro.uol.com.br/v2/transactions/notifications/%s?email=%s&token=%s"
+
+        is_dev = PaymentProcessor.get_backend_setting('testing', False)
+        if is_dev:
+            _TRANSACTION_CONSULT_URL = "https://ws.sandbox.pagseguro.uol.com.br/v2/transactions/"+\
+             "notifications/%s?email=%s&token=%s"
+
+        Payment = get_model('getpaid', 'Payment')
+        
+        token = PaymentProcessor.get_backend_setting('token')
+        email = PaymentProcessor.get_backend_setting('email')
+
+        url_consult = _TRANSACTION_CONSULT_URL % (params["notificationCode"], email, token)
+        resp = requests.get(url_consult)
+        logger.info("pagseguro notification: " + resp.text)
+
+        dom = parseString(resp.text.encode('utf-8'))
+        transactionNode = dom.getElementsByTagName("transaction")
+        code = dom.getElementsByTagName("code")[0].firstChild.nodeValue
+        if dom.getElementsByTagName("status"):
+            status_code = int(dom.getElementsByTagName("status")[0].firstChild.nodeValue)
+        
+        reference = ""
+        reference_node = dom.getElementsByTagName("reference")
+        if reference_node:
+            reference = reference_node[0].firstChild.nodeValue
+        
+        try:
+            payment = Payment.objects.get(external_id__exact=reference)
+        except Payment.DoesNotExist:
+            logger.error('Payment does not exist with external_id=%s' % reference)
+            return
+        
+        if status_code in (PagseguroTransactionStatus.AVAILABLE, PagseguroTransactionStatus.PAID):
+            logger.info("Updating payment" + str(payment.id))
+            amount = dom.getElementsByTagName("grossAmount")[0].firstChild.nodeValue
+            payment.amount_paid = Decimal(amount)
+            payment.paid_on = datetime.datetime.utcnow().replace(tzinfo=utc)
+            payment.change_status('paid')
+
+        elif status_code in (PagseguroTransactionStatus.CANCELED,
+                             PagseguroTransactionStatus.REFUNDED,
+                             PagseguroTransactionStatus.IN_DISPUTE):
+            payment.change_status('failed')
+
+
+    @staticmethod
+    def _get_view_full_url(request, view_name, args=None):
+        url = reverse(view_name, args=args)
+        return 'http://%s%s' % (request.get_host(), url)
```

Comment:
precisely `build_absolute_uri`, without domain kwarg to fetch domain from `get_domain` function